src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,53 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGR = 18;
+    private static final int MIN_SIZE = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
+    private void isAgeValid(User user) {
+        if (user.getAge() == null) {
+            throw new NullPointerException("Age can`t be null");
+        }
+
+        if (user.getAge() < MIN_AGR) {
+            throw new RegistrationException("Not valid age " + user.getAge()
+                    + " Min allowed age is " + MIN_AGR);
+        }
+    }
+
+    private void isLoginValid(User user) {
+        if (user.getLogin() == null) {
+            throw new NullPointerException("Login can`t be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
The method `isLoginValid(User user)` checks if the user exists in the storage before validating the login length. It's better to check the login length before checking the database to avoid unnecessary database interaction.